statement.k

```diff
@@ -185,10 +185,92 @@ imports CONFIGURATION
          rule if (false) S:Statement => .
          rule if (true)  S else S1:Statement => S
          rule if (false) S1:Statement else S => S
-
+/*
          rule for (A;B;C) S => A; while (B) {S C;} [structural]
          rule do S while (B) =>  S if(B) {do S while (B)} else {.Statements}  [structural]
-         rule while (B) S => if (B) {S while (B) S} else {.Statements}  [structural]
+         rule while (B) S => if (B) {S while (B) S} else {.Statements}  [structural]*/
+
+          rule <k> for (A;B;C) {S} => InitStmts(A;) if (B) {ExeWhile(C; S) for (A;B;C) {S}} else {EndLoop(.Statements)} ... </k>
+              <loopCon> true </loopCon>
+              <Continue> _=>false </Continue>
+              <funCond> true </funCond>
+
+         rule <k> for (A;B;C) {S} => Reset(1) ... </k> 
+              <loopCon> false </loopCon>
+
+         rule <k> for (A;B;C) {S} => . ... </k> 
+              <funCond> false </funCond>
+
+         rule <k> EndLoop(.Statements) => . ... </k>
+              <forInit> _=>true </forInit>
+
+         rule <k> InitStmts(S:Statement Ss:Statements) => InitStmt(S) ~> InitStmts(Ss) ... </k>
+         rule <k> InitStmt(S) => S ... </k>
+              <forInit> true=>false </forInit>
+         rule <k> InitStmt(S) => . ... </k>
+              <forInit> false </forInit>
+         rule <k> InitStmts(.Statements) => . ... </k>
+     
+              
+            
+
+         rule <k> do {S} while (B) =>  ExeWhile(S) if(B) {do {S} while (B)} else {.Statements} ... </k> 
+              <loopCon> true </loopCon>
+              <Continue> _=>false </Continue>
+              <funCond> true </funCond>
+
+        rule <k> do {S} while (B) => Reset(1) ... </k> 
+             <loopCon> false </loopCon>
+
+        rule <k> do {S} while (B) => . ... </k> 
+             <funCond> false </funCond>
+
+         
+
+               
+         rule <k> while (B) {S} => if (B) {ExeWhile(S) while (B) {S}} else {.Statements} ... </k> 
```

Comment:
In general, I suggest starting to organize the source file a bit better by adding something like "sections", explanation of rules (written as comments), maybe little examples etc. 

For example, it would be nice to have one "section" (for example, delineated with comments) for "for loop", one for "while loop" etc., and having all related rules close to each other so that it's easier for a user to read it. 

For an example, here is how I done it in my PHP semantics: https://github.com/dfilaretti/KPHP/blob/master/src/control-structures.k (by the way, you may also find it useful to see how I did `break` and `continue`). But notice that this is outdated, as now people prefer to write markdown-like comments - see https://github.com/runtimeverification/iele-semantics/blob/master/iele.md for an example. 

Notice that we don't have to write extensive documentation right now (we can do that later), what I'm suggesting is just that we start keeping that in mind and try to keep things in order and easier to understand.